packages/netlify-cms-backend-github/src/GraphQLAPI.js

```diff
@@ -0,0 +1,459 @@
+        query: queries.unpublishedBranchFile,
+        variables: {
+          owner,
+          name,
+          headExpression: `${metaData.branch}:${path}`,
+          baseExpression: `${this.branch}:${path}`,
+        },
+      });
+      if (!data.head.object) {
+        throw new EditorialWorkflowError('content is not under editorial workflow', true);
+      }
+      const result = {
+        metaData,
+        fileData: data.head.object.text,
+        isModification: !!data.base.object,
+      };
+      return result;
+    } else {
+      throw new EditorialWorkflowError('content is not under editorial workflow', true);
+    }
+  }
+
+  getBranchQualifiedName(branch) {
+    return `refs/heads/${branch}`;
+  }
+
+  getBranchQuery(branch) {
+    const { repo_owner: owner, repo_name: name } = this;
+
+    return {
+      query: queries.branch,
+      variables: {
+        owner,
+        name,
+        qualifiedName: this.getBranchQualifiedName(branch),
+      },
+    };
+  }
+
+  async getBranch(branch = this.branch) {
+    const { data } = await this.query({
+      ...this.getBranchQuery(branch),
+      fetchPolicy: CACHE_FIRST, // branch query will be updated by mutations
+    });
+    return data.repository.branch;
+  }
+
+  async patchRef(type, name, sha, opts = {}) {
+    if (type !== 'heads') {
+      return super.patchRef(type, name, sha, opts);
+    }
+
+    const force = opts.force || false;
+
+    const branch = await this.getBranch(name);
+    const { data } = await this.mutate({
+      mutation: mutations.updateBranch,
+      variables: {
+        input: { oid: sha, refId: branch.id, force },
+      },
+    });
+    return data.updateRef.branch;
+  }
+
+  async deleteBranch(branchName) {
+    const branch = await this.getBranch(branchName);
+    const { data } = await this.mutate({
+      mutation: mutations.deleteBranch,
+      variables: {
+        deleteRefInput: { refId: branch.id },
+      },
+      update: store => store.data.delete(defaultDataIdFromObject(branch)),
+    });
+
+    return data.deleteRef;
+  }
+
+  getPullRequestAndBranchQuery(branch, number) {
+    const { repo_owner: owner, repo_name: name } = this;
+
+    return {
+      query: queries.pullRequestAndBranch,
+      variables: { owner, name, number, qualifiedName: this.getBranchQualifiedName(branch) },
+    };
+  }
+
+  async pullRequestAndBranch(branch, number) {
+    const { data } = await this.query({
+      ...this.getPullRequestAndBranchQuery(branch, number),
+      fetchPolicy: CACHE_FIRST,
+    });
+
+    return data.repository;
+  }
+
+  async deleteUnpublishedEntry(collection, slug) {
+    try {
+      const contentKey = slug;
+      const branchName = this.generateBranchName(contentKey);
+
+      const metadata = await this.retrieveMetadata(contentKey);
+      const { branch, pullRequest } = await this.pullRequestAndBranch(
+        branchName,
+        metadata.pr.number,
+      );
+
+      const { data } = await this.mutate({
```

Comment:
On mutation instead of two requests in the base class (close pr, then delete branch)